planex/makesrpm.py

```diff
@@ -0,0 +1,160 @@
+    return parser.parse_known_args(argv)
+
+
+def parse_patchseries(series_file):
+    """
+    Parse series file and return the list of patches
+    """
+    patches = []
+    with open(series_file) as series:
+        for line in series:
+            line = line.partition('#')[0].strip()
+            if line:
+                patches.append(line)
+
+    return patches
+
+
+def setup_tmp_area():
+    """
+    Create temporary working area
+    """
+    tmp_dirpath = tempfile.mkdtemp()
+    tmp_specs = os.path.join(tmp_dirpath, 'SPECS')
+    tmp_build = os.path.join(tmp_dirpath, '_build')
+    tmp_sources = os.path.join(tmp_build, 'SOURCES')
+
+    os.makedirs(tmp_specs)
+    os.makedirs(tmp_build)
+    os.makedirs(tmp_sources)
+
+    return (tmp_dirpath, tmp_sources)
+
+
+def extract_patches(tmp_specfile, patchqueue_filters, patchqueue_path,
+                    tmp_sources, target):
+    """
+    Extract patches and inject them into the specfile
+    """
+    for line in fileinput.input(tmp_specfile, inplace=True):
+        if any([ext in line for ext in patchqueue_filters]):
+            tar = tarfile.open(patchqueue_path)
+            for mem in tar.getmembers():
+                mem.name = '%s-' % target + os.path.basename(mem.name)
+                tar.extract(mem, tmp_sources)
+            patches = parse_patchseries(os.path.join(tmp_sources,
+                                                     '%s-series' % target))
+            print "# Patches for %s" % target
+            for patch_num, patch in enumerate(patches):
+                if patch != "":
+                    print "Patch%s: %%{name}-%s" % (patch_num, patch)
+                    patch_num += 1
+        else:
+            print line,
+
+
+def get_command_line(intercepted_args, tmp_sources, tmp_specfile):
+    """
+    Return rpmbuild command line and arguments
+    """
+    cmd = ['rpmbuild']
+    if intercepted_args.quiet:
+        cmd.append('--quiet')
+    if intercepted_args.topdir is not None:
+        cmd.append('--define')
+        cmd.append('_topdir %s' % intercepted_args.topdir)
+    if intercepted_args.dist != "":
+        cmd.append('--define')
+        cmd.append('%%dist %s' % intercepted_args.dist)
+    cmd.append('--define')
+    cmd.append('_sourcedir %s' % tmp_sources)
+    cmd.append('-bs')
+    cmd.append(tmp_specfile)
+
+    return cmd
+
+
+def main(argv):
+    """
+    Entry point arguments: component.spec Source0 Source1 ... Patch0 Patch1 ...
+    The specfile must always be present and it has to be the first in the list
+    """
+    intercepted_args, passthrough_args = parse_args_or_exit(argv)
+    target = os.path.splitext(os.path.basename(passthrough_args[0]))[0]
+    specfile = passthrough_args[0]
+    tmp_dirpath, tmp_sources = setup_tmp_area()
+    tmp_specfile = os.path.join(tmp_dirpath, specfile)
+
+    try:
+        # Copy files to temporary working area
+        copyfile(specfile, tmp_specfile)
+        patchqueue = False
```

Comment:
You can get rid of these variables if you move extract_patches as described below.